### ISSUE TITLE
Unexpected end of stream when column is empty.

### ISSUE THREAD

**Issue by dgrr** [ClickHouse/ClickHouse#6611]
Hello,

My ClickHouse version is:
```
SELECT version()

┌─version()─┐
│ 19.11.2.7 │
└───────────┘
```

I don't know if the problem will have a specific fix. The error is caused because the last field isn't filled (it's nullable), but ClickHouse server returns an error of `unexpected end of stream`.
You can try by using:
```
CREATE TABLE xx
(
    `id` Int64, 
    `name` Nullable(String), 
    `price` Nullable(Float)
)
ENGINE = Memory
```

`echo -en "1\ta\t0.1\n2\tb\t" | ch -q 'insert into xx format TSV'`
The error will be `Code: 33. DB::Exception: Unexpected end of stream, while parsing value of Nullable type: (at row 2)` because the last field is empty. If `echo -en "1\ta\t0.1\n2\tb\t\n" | ch -q 'insert into xx format TSV'` is executed the query works. But that not the point here.

Thanks.

**Comment by alexey-milovidov** [comment]
That's the expected behaviour.

Empty string doesn't read as NULL. To put NULL into TSV file, you have to write `\N`:
```
echo -en "1\ta\t0.1\n2\tb\t\\N" | ch -q 'insert into xx format TSV'
```

**Comment by alexey-milovidov** [comment]
@tavplubix FYI that's about possible implementation of `input_format_tsv_empty_as_default` setting.

**Comment by filimonov** [comment]
Related / duplicate: https://github.com/yandex/ClickHouse/issues/469

**Comment by tavplubix** [comment]
Resolved in #7215
